pkg/sarif.go

```diff
@@ -149,33 +149,49 @@ type sarif210 struct {
 	Runs    []run  `json:"runs"`
 }
 
+func maxOffset(x, y int) int {
+	if x < y {
+		return y
+	}
+	return x
+}
+
 func detailToRegion(details *checker.CheckDetail) region {
 	var reg region
 	var snippet *text
 	if details.Msg.Snippet != "" {
 		snippet = &text{Text: details.Msg.Snippet}
 	}
+
+	// https://docs.oasis-open.org/sarif/sarif/v2.0/csprd02/sarif-v2.0-csprd02.html.
+	// "3.30.1 General".
+	// Line numbers > 0.
+	// byteOffset and charOffset >= 0.
 	switch details.Msg.Type {
 	default:
 		panic("invalid")
 	case checker.FileTypeURL:
+		line := maxOffset(1, details.Msg.Offset)
 		reg = region{
-			StartLine: &details.Msg.Offset,
+			StartLine: &line,
 			Snippet:   snippet,
 		}
 	case checker.FileTypeNone:
 		// Do nothing.
 	case checker.FileTypeSource:
+		line := maxOffset(1, details.Msg.Offset)
 		reg = region{
-			StartLine: &details.Msg.Offset,
+			StartLine: &line,
 			Snippet:   snippet,
 		}
 	case checker.FileTypeText:
+		// Offset of 0 is acceptable here.
 		reg = region{
 			CharOffset: &details.Msg.Offset,
```

Comment:
To be safe, should we specify `maxOffset(0, Offset)`? In case `Offset` is < 0.

Reply:
`<0` means a bug in our checks code. If I do that it will go unnoticed. Shall I add a `panic` instead?

Reply:
Yeah, or atleast a log. Looks like right now, we silently fail which is not good.